### PR TITLE
Let pre-existing sourceRoot through

### PR DESCRIPTION
**Pull request by jasonholloway**
Hi there, 

Source roots were being overwritten, even when in place (as mentioned a while back: #11), which was mucking up my attempts at getting Browserify to work nicely on my project. Like the poster in the linked issue, I can't think of any good reason why this would be the case - hence this speculative PR.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ var mold = require('mold-source-map')
   , fs = require('fs');
 
 function separate(src, file, root, base, url) {
-  src.sourceRoot(root || '');
+  src.sourceRoot(root || (src.sourcemap.getProperty('sourceRoot') || ''));
   if (base) {
```
Comment (abritinthebay) [inline index.js]: Could this not just be written as `src.sourceRoot(root || src.sourcemap.getProperty('sourceRoot') || '');` ? Or am I missing a logic issue here?

File: test/exorcist.js [hunk 2]
```diff
@@ -39,7 +40,7 @@ test('\nwhen piping a bundle generated with browserify through exorcist without
       t.equal(map.sources[0].indexOf(base), 0, 'uses absolute source paths')
       t.equal(map.sourcesContent.length, 4, 'includes 4 source contents')
       t.equal(map.mappings.length, 106, 'maintains mappings')
-      t.equal(map.sourceRoot, '', 'leaves source root an empty string')
+      t.equal(map.sourceRoot, '', 'if source root missing, emplaces empty string')
```
Comment (abritinthebay) [inline test/exorcist.js]: "emplace", goddamn its nice to see someone who knows a dictionary. Top marks just for word usage! 👍

File: index.js (hunk 1 above)
Comment (jasonholloway) [inline index.js]: Err, you may have a point, possibly.

**Comment by bendrucker** [comment]
Hey, I wanted to go ahead and get this merged. @jasonholloway the code looks good to me, is there anything you wanted to add/update? If not I'll check it out locally and get it in.

**Comment by jasonholloway** [comment]
All should be ok as far as I'm aware - thanks!

**Comment by bendrucker** [comment]
@thlorenz Can I get `npm` access so I can publish this as 0.5.0?

**Pull request closed without merging**